doc/design.md

```diff
@@ -94,31 +95,41 @@ Covers:
 
 Needs: impl, utest, itest
 
-## `UserInformation` Caches User Roles
-`dsn~user-information-caches-user-roles`
-
-The `UserInformation` caches the roles a user has.
+## Query Rewriter Determines User Roles
 
-Rationale:
+## Query Rewriter Identifies Protected Tables
+`dsn~query-rewriter-identifies-protected-tables~1`
 
-This speeds up subsequent row access checks because the users roles need to be determined only once.
+The Query Rewriter identifies a table as protected with row-level security, if that table has a column named `exa_row_roles` or a column named  `exa_row_tenants` or both.
 
 Covers:
 
-* `req~user-roles~1`
+* `req~rows-users-are-allowed-to-read~1`
+* `req~tables-with-tenants-restrictions~1`
 
-Needs: impl, utest
+Needs: impl, utest, itest
 
-## Query Rewriter Determines User Roles
+## Query Rewriter Identifies Unprotected Tables
+`dsn~query-rewriter-identifies-unprotected-tables~1`
 
-## Query Rewriter Identifies Protected Tables
-`dsn~query-rewriter-identifies-protected-tables~1`
+The Query Rewriter identifies a table as unprotected, if that table does not have a column named `exa_row_roles` or `exa_row_tenants`.
+The Query Rewriter does not modify an unprotected table.
 
-The Query Rewriter identifies a table as protected with row-level security, if that table has a column named `exa_row_roles`.
+Covers:
+
+* `req~unprotected-tables~1`
+
+Needs: impl, utest, itest
+
+## Query Rewriter Chooses a Way to Treat Protected Tables
```

Comment:
Follow-up issue. See comment in system requirements.

Reply:
fixed